src/dmd/statementsem.d

```diff
@@ -219,6 +219,28 @@ private extern (C++) final class StatementSemanticVisitor : Visitor
             }
         }
 
+        /* Flatten them in place
+         */
+        void flatten(Statements* statements)
+        {
+            for (size_t i = 0; i < statements.length;)
```

Comment:
Could be `foreach (i; 0 ... statements.length)` like the code did before.

Reply:
No, because `i` isn't always incremented.